toast-tk-web-agent/src/main/java/io/toast/tk/agent/web/ScriptInjector.java

```diff
@@ -29,27 +30,28 @@
 	private JFrame frmOpt; 
 	private IAgentServer server;
 	private UriChangeListener uriChangeListener;
-	private WebConfigProvider configProvider;
-	
+	private AgentConfigProvider configProvider;
+	private volatile boolean recording;
+
 	@Inject
-	public ScriptInjector(IAgentServer server, UriChangeListener uriChangeListener, WebConfigProvider configProvider){
+	public ScriptInjector(IAgentServer server, UriChangeListener uriChangeListener, AgentConfigProvider configProvider){
 		this.server = server;
 		this.uriChangeListener = uriChangeListener;
 		this.configProvider = configProvider;
+		this.recording = true;
 	}
 	
 	private Thread initInjectionRetryThread() {
 		return new Thread(new Runnable() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1604)